### ISSUE TITLE
[Bug]: M1 Mac, interactive fails in jupyter notebook and jupyter lab

### ISSUE THREAD

**Issue by shiba24** [matplotlib/matplotlib#22545]
### Bug summary

Hi, the interactive figure in jupyter lab on Mac M1 is broken in my environment.
I think this is the same issue. Could you help me how to solve it?

https://stackoverflow.com/questions/70466222/m1-mac-python-assertion-failure-and-crashes-when-using-osx-as-backend-in-matplo

### Code for reproduction

```python
# Open jupyterlab

%matplotlib osx
```

### Actual outcome

Endless

```
2022-02-23 19:00:11.233 Python[7544:10628886] *** Assertion failure in +[NSEvent otherEventWithType:location:modifierFlags:timestamp:windowNumber:context:subtype:data1:data2:], NSEvent.m:740
...
```

### Expected outcome

Interactive figures:

https://matplotlib.org/stable/users/explain/interactive.html

### Operating system

Mac OS Monterey 12.2 (M1 2020)

### Matplotlib Version

3.4.3

### Matplotlib Backend

module://matplotlib_inline.backend_inline

### Python version

3.9.9

### Jupyter version

jupyter lab 3.2.1

### Installation

pip

### [Update]

I confirmed the same error in jupyter notebook. version info:

```
jupyter notebook --version
6.4.5

jupyter lab --version
3.2.1

ipython --version
7.28.0
```

**Comment by dstansby** [comment]
I can confirm this on an M1 mac.

**Comment by dstansby** [comment]
```python
import matplotlib
matplotlib.use('macosx')
import matplotlib.pyplot as plt

plt.plot(0, 0)
plt.show()
```
Run in a terminal works fine, so this bug seems to involve interaction with jupyter (lab?)

**Comment by shiba24** [comment]
Thank you for the confirmation. For visibility, I posted the same issue in the jupyterlab repo too.

**Comment by greglucas** [comment]
This may actually be an ipython integration issue. Can you post which version of ipython you're using?

Also post whether this works in the following three situations (it looks like 1 should work):
1. `python script.py`
2. `ipython script.py`
3. `ipython -i script.py`

**Comment by dstansby** [comment]
`import matplotlib; matplotlib.use('macosx')` works fine for me in all those situations above. I'm not sure how to test `%matplotlib osx` (the code causing the orignial issue) in scripts/IPython though?

**Comment by greglucas** [comment]
You should get a prompt if you open ipython without specifying a script. You could

```bash
> ipython

[1]: %matplotlib osx`
[2]: %run script.py
```

**Comment by dstansby** [comment]
Running `%matplotlib osx` in `ipython` works fine, so it seems to be an issue specific with Jupyter notebook integration. I can reproduce running in a plain `jupyter notebbok`, and it looks like from the issue at the top it's an issue with `jupyter lab` too.

**Comment by shiba24** [comment]
I confirmed the same error in jupyter notebook. Update the title and description of the issue.

```
jupyter notebook --version
6.4.5

jupyter lab --version
3.2.1
```

**Comment by greglucas** [comment]
Interesting that ipython works, but Jupyter does not because I thought they were using the ipython event loop integrations. That is a useful datapoint! I don't think we control any of the "%matplotlib" magic, so I'm still not sure that this is a bug with us, versus more likely upstream in Jupyter.

**Comment by shiba24** [comment]
Is there any way to drill down the error log from matplotlib side, to make sure it's from jupyter? I'm happy to provide further logs etc.

Jupyter lab closed the issue that I posted. Even if we'd re-open it (or open another in jupyter notebook repo?), probably additional info would help them I think.

(Obviously I'm not the member of the matplotlib nor jupyter teams, so if I'm not the best person for doing things, will leave this to anyone appropriate..)

**Comment by tacaswell** [comment]
> Interesting that ipython works, but Jupyter does not because I thought they were using the ipython event loop integrations.

IIRC jupyter has its own copy of the input hooks (because they need to integrate with torando not with prompt toolkit)

**Comment by greglucas** [comment]
You can also try launching with some more info in the logs to see if anything obvious gets spit out there.
`jupyter-notebook --debug` (I don't know what `%gui osx` does versus `%matplotlib osx`, but that might be worth a shot too to see if that fails).
My guess is we should ping that other issue for a reopen, as that group will know more about the Jupyter integrations.

**Comment by tacaswell** [comment]
They have re-opened the issue on jlab, I'm going to close this issue.  Please ping to have re-opened (or open a new issue) if it is indeed Matplotlib's bug!

**Comment by LumosPotter** [comment]
is there any solution for this, I also encounter this bug and have no idea of this

**Comment by shiba24** [comment]
@LumosPotter  I guess the fix for this issue has been merged recently: https://github.com/ipython/ipykernel/pull/1237

**Comment by ianthomas23** [comment]
> @LumosPotter I guess the fix for this issue has been merged recently: [ipython/ipykernel#1237](https://github.com/ipython/ipykernel/pull/1237)

Yes, this should all work as expected now with `ipykernel >= 6.29.5`. If it is not, please open a new issue.
